### PR TITLE
Eliminate last "nitpicky" exceptions

### LINKED ISSUE #9411: Undefined token in the manual
The following list of 282 warnings was obtained by running `SPHINXWARNOPT="-w/tmp/warn.log" dune build @refman-html` after changing

https://github.com/coq/coq/blob/6994539744e4ffaa4f622c8bccc66276e445ae9a/doc/sphinx/conf.py#L180-L181

to `True`:

```
generalized-rewriting.rst:532: WARNING: 'token' reference target not found: orientation
generalized-rewriting.rst:532: WARNING: 'token' reference target not found: occs
generalized-rewriting.rst:532: WARNING: 'token' reference target not found: tactic
generalized-rewriting.rst:574: WARNING: 'token' reference target not found: Aeq
generalized-rewriting.rst:574: WARNING: 'token' reference target not found: ST
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: s
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: c
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: c
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: s
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: s
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: s
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: u
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: s
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: s
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: s
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: s
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: s
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: s
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: s
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: s
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: s
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: hintdb
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: c
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: redexpr
generalized-rewriting.rst:763: WARNING: 'token' reference target not found: c
generalized-rewriting.rst:775: WARNING: 'token' reference target not found: s
generalized-rewriting.rst:775: WARNING: 'token' reference target not found: id
generalized-rewriting.rst:775: WARNING: 'token' reference target not found: u
generalized-rewriting.rst:775: WARNING: 'token' reference target not found: s
generalized-rewriting.rst:775: WARNING: 'token' reference target not found: u
generalized-rewriting.rst:775: WARNING: 'token' reference target not found: s
generalized-rewriting.rst:775: WARNING: 'token' reference target not found: s
generalized-rewriting.rst:775: WARNING: 'token' reference target not found: bu
generalized-rewriting.rst:775: WARNING: 'token' reference target not found: td
generalized-rewriting.rst:775: WARNING: 'token' reference target not found: i
generalized-rewriting.rst:775: WARNING: 'token' reference target not found: o
generalized-rewriting.rst:813: WARNING: 'token' reference target not found: s

program.rst:286: WARNING: 'token' reference target not found: tactic
program.rst:310: WARNING: 'token' reference target not found: tactic
program.rst:314: WARNING: 'token' reference target not found: tactic

ring.rst:318: WARNING: 'token' reference target not found: ltac
ring.rst:318: WARNING: 'token' reference target not found: ltac
ring.rst:318: WARNING: 'token' reference target not found: ltac
ring.rst:318: WARNING: 'token' reference target not found: ltac
ring.rst:349: WARNING: 'token' reference target not found: ltac
ring.rst:344: WARNING: 'token' reference target not found: ltac
ring.rst:355: WARNING: 'token' reference target not found: ltac
ring.rst:352: WARNING: 'token' reference target not found: ltac
ring.rst:360: WARNING: 'token' reference target not found: ltac
ring.rst:358: WARNING: 'token' reference target not found: ltac
ring.rst:374: WARNING: 'token' reference target not found: ltac
ring.rst:363: WARNING: 'token' reference target not found: ltac

universe-polymorphism.rst:379: WARNING: 'token' reference target not found: ord
universe-polymorphism.rst:381: WARNING: 'token' reference target not found: ord

gallina-extensions.rst:601: WARNING: 'token' reference target not found: decrease_annot
gallina-extensions.rst:717: WARNING: 'token' reference target not found: ident_equation
gallina-extensions.rst:872: WARNING: 'token' reference target not found: module
gallina-extensions.rst:880: WARNING: 'token' reference target not found: module
gallina-extensions.rst:882: WARNING: 'token' reference target not found: module
gallina-extensions.rst:882: WARNING: 'token' reference target not found: module
gallina-extensions.rst:936: WARNING: 'token' reference target not found: module
gallina-extensions.rst:940: WARNING: 'token' reference target not found: module
gallina-extensions.rst:942: WARNING: 'token' reference target not found: module
gallina-extensions.rst:942: WARNING: 'token' reference target not found: module
gallina-extensions.rst:1836: WARNING: 'token' reference target not found: rename
gallina-extensions.rst:2172: WARNING: 'token' reference target not found: names

ltac.rst:18: WARNING: 'token' reference target not found: cpattern
ltac.rst:18: WARNING: 'token' reference target not found: atomic_tactic
ltac.rst:18: WARNING: 'token' reference target not found: cpattern
ltac.rst:18: WARNING: 'token' reference target not found: cpattern
ltac.rst:48: WARNING: 'token' reference target not found: tactic
ltac.rst:48: WARNING: 'token' reference target not found: tactic
ltac.rst:48: WARNING: 'token' reference target not found: tactic
ltac.rst:48: WARNING: 'token' reference target not found: tactic
ltac.rst:48: WARNING: 'token' reference target not found: tactic
ltac.rst:48: WARNING: 'token' reference target not found: tactic
ltac.rst:48: WARNING: 'token' reference target not found: tactic
ltac.rst:48: WARNING: 'token' reference target not found: tactic
ltac.rst:48: WARNING: 'token' reference target not found: tactic
ltac.rst:69: WARNING: 'token' reference target not found: tactic
ltac.rst:69: WARNING: 'token' reference target not found: tactic
ltac.rst:69: WARNING: 'token' reference target not found: tactic
ltac.rst:69: WARNING: 'token' reference target not found: tactic
ltac.rst:69: WARNING: 'token' reference target not found: tactic
ltac.rst:73: WARNING: 'token' reference target not found: tactic
ltac.rst:73: WARNING: 'token' reference target not found: tactic
ltac.rst:73: WARNING: 'token' reference target not found: tactic
ltac.rst:73: WARNING: 'token' reference target not found: tactic
ltac.rst:73: WARNING: 'token' reference target not found: tactic
ltac.rst:154: WARNING: 'token' reference target not found: redexpr
ltac.rst:154: WARNING: 'token' reference target not found: atomic_tactic
ltac.rst:154: WARNING: 'token' reference target not found: cpattern
ltac.rst:154: WARNING: 'token' reference target not found: cpattern
ltac.rst:154: WARNING: 'token' reference target not found: cpattern
ltac.rst:154: WARNING: 'token' reference target not found: cpattern
ltac.rst:154: WARNING: 'token' reference target not found: cpattern
ltac.rst:154: WARNING: 'token' reference target not found: cpattern
ltac.rst:154: WARNING: 'token' reference target not found: cpattern
ltac.rst:154: WARNING: 'token' reference target not found: cpattern
ltac.rst:770: WARNING: 'token' reference target not found: cpattern
ltac.rst:786: WARNING: 'token' reference target not found: cpattern
ltac.rst:797: WARNING: 'token' reference target not found: cpattern
ltac.rst:817: WARNING: 'token' reference target not found: cpattern
ltac.rst:826: WARNING: 'token' reference target not found: cpattern
ltac.rst:835: WARNING: 'token' reference target not found: cpattern
ltac.rst:871: WARNING: 'token' reference target not found: cpattern
ltac.rst:874: WARNING: 'token' reference target not found: cpattern
ltac.rst:909: WARNING: 'token' reference target not found: cpattern
ltac.rst:920: WARNING: 'token' reference target not found: cpattern
ltac.rst:971: WARNING: 'token' reference target not found: redexpr
ltac.rst:973: WARNING: 'token' reference target not found: redexpr

proof-handling.rst:153: WARNING: 'token' reference target not found: tactic
proof-handling.rst:178: WARNING: 'token' reference target not found: collection1
proof-handling.rst:178: WARNING: 'token' reference target not found: collection2
proof-handling.rst:183: WARNING: 'token' reference target not found: collection1
proof-handling.rst:183: WARNING: 'token' reference target not found: collection2
proof-handling.rst:188: WARNING: 'token' reference target not found: collection
proof-handling.rst:193: WARNING: 'token' reference target not found: collection
proof-handling.rst:205: WARNING: 'token' reference target not found: expression
proof-handling.rst:208: WARNING: 'token' reference target not found: expression
proof-handling.rst:223: WARNING: 'token' reference target not found: expression

ssreflect-proof-language.rst:1342: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:1353: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:1362: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:1548: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:1558: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:1567: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:1602: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:1602: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2057: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2061: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2061: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2061: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2061: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2076: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2076: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2168: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2168: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2168: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2168: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2242: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2242: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2312: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2316: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2329: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2359: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2461: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:2470: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:3478: WARNING: 'token' reference target not found: r_items
ssreflect-proof-language.rst:3986: WARNING: 'token' reference target not found: c_patterns
ssreflect-proof-language.rst:5317: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5394: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5394: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5394: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5394: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5408: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5408: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5408: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5408: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5452: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5456: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5460: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5460: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5464: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5464: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5464: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5468: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5472: WARNING: 'token' reference target not found: tactic
ssreflect-proof-language.rst:5472: WARNING: 'token' reference target not found: tactic

tactics.rst:42: WARNING: 'token' reference target not found: tactic
tactics.rst:42: WARNING: 'token' reference target not found: tactic
tactics.rst:1374: WARNING: 'token' reference target not found: tactic
tactics.rst:1392: WARNING: 'token' reference target not found: tactic
tactics.rst:1406: WARNING: 'token' reference target not found: tactic
tactics.rst:1446: WARNING: 'token' reference target not found: tactic
tactics.rst:1446: WARNING: 'token' reference target not found: tactic
tactics.rst:1449: WARNING: 'token' reference target not found: tactic
tactics.rst:1449: WARNING: 'token' reference target not found: tactic
tactics.rst:1454: WARNING: 'token' reference target not found: tactic
tactics.rst:1454: WARNING: 'token' reference target not found: tactic
tactics.rst:2129: WARNING: 'token' reference target not found: terms
tactics.rst:2297: WARNING: 'token' reference target not found: terms
tactics.rst:2782: WARNING: 'token' reference target not found: tactic
tactics.rst:2784: WARNING: 'token' reference target not found: tactic
tactics.rst:2802: WARNING: 'token' reference target not found: tactic
tactics.rst:2892: WARNING: 'token' reference target not found: tactic
tactics.rst:2894: WARNING: 'token' reference target not found: tactic
tactics.rst:3551: WARNING: 'token' reference target not found: tactic
tactics.rst:3560: WARNING: 'token' reference target not found: tactic
tactics.rst:3562: WARNING: 'token' reference target not found: tactic
tactics.rst:3565: WARNING: 'token' reference target not found: clause
tactics.rst:3567: WARNING: 'token' reference target not found: clause
tactics.rst:3589: WARNING: 'token' reference target not found: tactic
tactics.rst:3592: WARNING: 'token' reference target not found: tactic
tactics.rst:3592: WARNING: 'token' reference target not found: tactic
tactics.rst:3641: WARNING: 'token' reference target not found: hint_definition
tactics.rst:3643: WARNING: 'token' reference target not found: hint_definition
tactics.rst:3649: WARNING: 'token' reference target not found: hint_definition
tactics.rst:3656: WARNING: 'token' reference target not found: hint_definition
tactics.rst:3763: WARNING: 'token' reference target not found: tactic
tactics.rst:3766: WARNING: 'token' reference target not found: tactic
tactics.rst:3793: WARNING: 'token' reference target not found: regexp
tactics.rst:4007: WARNING: 'token' reference target not found: tactic
tactics.rst:4021: WARNING: 'token' reference target not found: tactic
tactics.rst:4076: WARNING: 'token' reference target not found: tactic
tactics.rst:4079: WARNING: 'token' reference target not found: tactic
tactics.rst:4146: WARNING: 'token' reference target not found: tactic
tactics.rst:4156: WARNING: 'token' reference target not found: tactic
tactics.rst:4158: WARNING: 'token' reference target not found: tactic
tactics.rst:4528: WARNING: 'token' reference target not found: x

vernacular-commands.rst:203: WARNING: 'token' reference target not found: convtactic
vernacular-commands.rst:273: WARNING: 'token' reference target not found: term_pattern
vernacular-commands.rst:275: WARNING: 'token' reference target not found: term_pattern
vernacular-commands.rst:280: WARNING: 'token' reference target not found: term_pattern_string
vernacular-commands.rst:282: WARNING: 'token' reference target not found: term_pattern_string
vernacular-commands.rst:282: WARNING: 'token' reference target not found: term_pattern
vernacular-commands.rst:292: WARNING: 'token' reference target not found: term_pattern_string
vernacular-commands.rst:292: WARNING: 'token' reference target not found: term_pattern_string
vernacular-commands.rst:297: WARNING: 'token' reference target not found: term_pattern_string
vernacular-commands.rst:297: WARNING: 'token' reference target not found: term_pattern_string
vernacular-commands.rst:302: WARNING: 'token' reference target not found: term_pattern_string
vernacular-commands.rst:302: WARNING: 'token' reference target not found: term_pattern_string
vernacular-commands.rst:377: WARNING: 'token' reference target not found: term_pattern
vernacular-commands.rst:620: WARNING: 'token' reference target not found: dirpath
vernacular-commands.rst:622: WARNING: 'token' reference target not found: dirpath
vernacular-commands.rst:622: WARNING: 'token' reference target not found: dirpath
vernacular-commands.rst:622: WARNING: 'token' reference target not found: dirpath
vernacular-commands.rst:652: WARNING: 'token' reference target not found: dirpath
vernacular-commands.rst:652: WARNING: 'token' reference target not found: dirpath
vernacular-commands.rst:732: WARNING: 'token' reference target not found: dirpath
vernacular-commands.rst:734: WARNING: 'token' reference target not found: dirpath
vernacular-commands.rst:740: WARNING: 'token' reference target not found: dirpath
vernacular-commands.rst:744: WARNING: 'token' reference target not found: dirpath
vernacular-commands.rst:746: WARNING: 'token' reference target not found: dirpath
vernacular-commands.rst:752: WARNING: 'token' reference target not found: dirpath
vernacular-commands.rst:765: WARNING: 'token' reference target not found: dirpath
vernacular-commands.rst:767: WARNING: 'token' reference target not found: dirpath
vernacular-commands.rst:798: WARNING: 'token' reference target not found: dirpath
vernacular-commands.rst:800: WARNING: 'token' reference target not found: dirpath
vernacular-commands.rst:927: WARNING: 'token' reference target not found: command
vernacular-commands.rst:929: WARNING: 'token' reference target not found: command
vernacular-commands.rst:933: WARNING: 'token' reference target not found: command
vernacular-commands.rst:935: WARNING: 'token' reference target not found: command
vernacular-commands.rst:939: WARNING: 'token' reference target not found: command
vernacular-commands.rst:941: WARNING: 'token' reference target not found: command
vernacular-commands.rst:954: WARNING: 'token' reference target not found: command
vernacular-commands.rst:956: WARNING: 'token' reference target not found: command
vernacular-commands.rst:956: WARNING: 'token' reference target not found: command
vernacular-commands.rst:1109: WARNING: 'token' reference target not found: level
vernacular-commands.rst:1134: WARNING: 'token' reference target not found: level
vernacular-commands.rst:1156: WARNING: 'token' reference target not found: convtactic
vernacular-commands.rst:1158: WARNING: 'token' reference target not found: convtactic
vernacular-commands.rst:1179: WARNING: 'token' reference target not found: command
vernacular-commands.rst:1179: WARNING: 'token' reference target not found: command

syntax-extensions.rst:313: WARNING: 'token' reference target not found: symbol
syntax-extensions.rst:313: WARNING: 'token' reference target not found: modifier
syntax-extensions.rst:317: WARNING: 'token' reference target not found: symbol
syntax-extensions.rst:317: WARNING: 'token' reference target not found: modifier
syntax-extensions.rst:673: WARNING: 'token' reference target not found: constr
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: tactic
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: reference
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: hyp
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: hyp_list
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: ne_hyp_list
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: constr
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: uconstr
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: constr_list
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: ne_constr_list
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: integer_list
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: ne_integer_list
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: int_or_var
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: int_or_var_list
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: ne_int_or_var_list
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: tactic
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: tactic0
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: tactic1
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: tactic2
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: tactic3
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: tactic4
syntax-extensions.rst:1702: WARNING: 'token' reference target not found: tactic5
syntax-extensions.rst:1703: WARNING: 'token' reference target not found: level
syntax-extensions.rst:1703: WARNING: 'token' reference target not found: tactic
```

A large number of them could be fixed by defining the token `tactic`. In particular, I currently do not like at all the tokens that are exported by the ltac grammar (`expr` is not very expressive, is it the one that should be renamed `tactic`? `atom` isn't very expressive either; `top` is quite weird too).

### PR DESCRIPTION
**Pull request by jfehrle**
Fixes: #9411

### CODE REVIEW

**Comment by ppedrot** [comment]
@Zimmi48 you should probably assign and merge.

File: doc/sphinx/conf.py [hunk 1]
```diff
@@ -187,10 +187,6 @@ def setup(app):
 nitpicky = True
 
 nitpick_ignore = [ ('token', token) for token in [
-    'induction_clause',
-    'where',
-    'bindings_with_parameters',
-    'destruction_arg'
 ]]
```
Comment (Zimmi48) [inline doc/sphinx/conf.py:190]: Remove this entirely. We do not want to ease the re-introduction of exceptions.

**Review by Zimmi48 — APPROVED** [review]
Wow, that's wonderful! Sorry for not noticing the PR!

File: doc/sphinx/conf.py (hunk 1 above)
Comment (Zimmi48) [reply doc/sphinx/conf.py:190]: Sorry, `nitpicky = True` should stay. What should be removed is the entire `nitpick_ignore` definition.

**Comment by jfehrle** [comment]
Got it, please merge. Theo, would you also review/merge #17214, #17230 and #17229 after I rebase it?

**Review by Zimmi48 — APPROVED** [review]
@coqbot merge now

**Pull request merged**